functional/keylime_tenant-ima-signature-sanity/test.sh

```diff
@@ -0,0 +1,126 @@
+        rlRun "limeWaitForAgentStatus ${AGENT_ID} 'Get Quote'"
+        rlRun -s "keylime_tenant -c cvlist"
+        rlAssertGrep "{'code': 200, 'status': 'Success', 'results': {'uuids':.*'${AGENT_ID}'" $rlRun_LOG -E
+    rlPhaseEnd
+
+    rlPhaseStartTest "Test that IMA key verification using GPG signature fails for an invalid key"
+        rlRun "gpg --verify signature-gpg-fake.sig allowlist.txt"
+        rlRun -s "keylime_tenant -v 127.0.0.1 -t 127.0.0.1 -u ${AGENT_ID} --allowlist allowlist.txt --exclude excludelist.txt -f excludelist.txt --sign_verification_key ${limeIMAPublicKey} --signature-verification-key-sig signature-gpg-fake.sig --signature-verification-key-sig-key gpg-key-fake.pub -c update" 1
+        rlAssertGrep "WARNING - Unable to verify signature" $rlRun_LOG
+    rlPhaseEnd
+
+    rlPhaseStartTest "Verify IMA key using a downloaded GPG signature and key"
+        rlRun "cp ${limeIMAPublicKey} ."
+        rlRun "keylime_tenant -v 127.0.0.1 -t 127.0.0.1 -u ${AGENT_ID} --allowlist allowlist.txt --exclude excludelist.txt -f excludelist.txt --signature-verification-key-url 'http://localhost:8000/x509_evm.pem' --signature-verification-key-sig-url 'http://localhost:8000/signature-gpg-genuine.sig' --signature-verification-key-sig-url-key gpg-key-genuine.pub -c update"
+        rlRun "limeWaitForAgentStatus ${AGENT_ID} 'Get Quote'"
+        rlRun -s "keylime_tenant -c cvlist"
+        rlAssertGrep "{'code': 200, 'status': 'Success', 'results': {'uuids':.*'${AGENT_ID}'" $rlRun_LOG -E
+    rlPhaseEnd
+
+    rlPhaseStartTest "Test that IMA key verification using downloaded GPG signature fails for an invalid key"
+        rlRun "cp ${limeIMAPublicKey} ."
+        rlRun -s "keylime_tenant -v 127.0.0.1 -t 127.0.0.1 -u ${AGENT_ID} --allowlist allowlist.txt --exclude excludelist.txt -f excludelist.txt --signature-verification-key-url 'http://localhost:8000/x509_evm.pem' --signature-verification-key-sig-url 'http://localhost:8000/signature-gpg-fake.sig' --signature-verification-key-sig-url-key gpg-key-fake.pub -c update" 1
+        rlAssertGrep "WARNING - Unable to verify signature" $rlRun_LOG
+    rlPhaseEnd
+
+    rlPhaseStartCleanup "Do the keylime cleanup"
+        rlRun "kill $HTTP_PID"
+        rlRun "limeStopAgent"
+        rlRun "limeStopRegistrar"
+        rlRun "limeStopVerifier"
+        limeLogfileSubmit $(limeVerifierLogfile)
+        limeLogfileSubmit $(limeRegistrarLogfile)
+        limeLogfileSubmit $(limeAgentLogfile)
+        if limeTPMEmulated; then
+            rlRun "limeStopIMAEmulator"
+            limeLogfileSubmit $(limeIMAEmulatorLogfile)
+            rlRun "limeStopTPMEmulator"
+            rlServiceRestore tpm2-abrmd
+        fi
+        limeClearData
+        limeRestoreConfig
+        limeExtendNextExcludelist ${TESTDIR}
+        rlRun "rm -rf ${TmpDir}"
+        #delete of gpg created gpg keys
+        rlRun "rm -rf /root/.gnupg/"
```

Comment:
You should be using `rlFileBackup` and `rlFileRestore`.
E.g. in a Setup phase backup existing content with `rlFileBackup --clean --missing-ok ~/.gnupg` and delete this dir.
In Cleanup call `rlFileRestore`.
One thing to keep in mine, `rlFileRestore` ends with an error if nothing has been backed up, e.g. when the original dir didn't exist and therefore was not backed up. What I usually do is to make sure I backup at least something. If there is no such need in a test, I do backup of `/etc/hosts` or similar file, just to have something to restore.

Reply:
I find out, that defautly on system aren't any pgp keys, so I decided to only delete dir with keys.

Reply:
Tests should try to return a system to a previous state as much as possible. Those keys are not present by default but maybe some keys will be installed by some setup task in the future. We do not know all the details about all test environments we are using, better to take precautions.

Reply:
I have just noticed that we are doing
`export GNUPGHOME=${TmpDir}`
and therefore we should not need to take care of `~/.gnupg`, I believe there shouldn't be anything created by the test. But I didn't double check it.

Reply:
I think that private keys, are still stored in `~/.gnupg`.

Reply:
`:: [ 13:35:04 ] :: [  BEGIN   ] :: Running 'gpg --batch --pinentry-mode=loopback --passphrase '' --generate-key gpg.script'
gpg: Generating a basic OpenPGP key
gpg: revocation certificate stored as '/root/.gnupg/openpgp-revocs.d/8CB01183C12CC3E17BCD5AD7FB5ABFA944ED27CC.rev'
gpg: done`